### DISCUSSION TITLE
Runs for around 15s then resets. Using Berry HASPmota and LVGL

### DISCUSSION THREAD

**Discussion by richardsg307** [arendst/Tasmota#20397] [Q&A] [answered]
I've nearly completed a Boiler monitor using a Lilygo T-Display-S3 ESP32, Berry, HASPmota and LVGL. Had it working on the bench with 2 DS18B20s and a button to represent Boiler Demand configured as a Switch Mode 2. It has 3 screens which I can scroll through using one of the buttons on the Lilygo board.
I just completed the wiring loom for the boiler with 3 DS18B20s and 2 digital inputs coming from 2 opto isolated mains detectors. I powered down on the bench and plugged in the wiring loom and powered back up. It now goes into a loop where it reboots about every 20 seconds! It now does this if the bench sensors are plugged back in and if nothing is plugged in.
Now during that time most things work. I can see the screens and scroll between them. The WiFi Arcs show connection to WiFi but the clock never displays the time. Two of the temperature sensors (new loom) work and report on my screen (I hadn't coded in the third to display). In the 12 seconds or so it shows connection to the WiFi I cannot access the Web interface. I have it plugged into the USB and using a serial monitor I get this in the 20 second loop:
```
Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42160a72
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x128
load:0x403c9700,len:0xb90
load:0x403cc700,len:0x28c8
SHA-256 comparison failed:
Calculated: f8f8dc53afbb7f195b56bb611a6f4f83f7d8aebcec67bad654d2c5b3322a2d94
Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
Attempting to boot anyway...
entry 0x403c98b8

00:00:00.000 CMD: Using USB CDC
00:00:00.001 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.035 UFS: FlashFS mounted with 12556 kB free
00:00:00.045 CFG: Loaded from File, Count 182
00:00:00.051 QPC: Count 1
00:00:00.055 I2C: Bus1 using GPIO17(SCL) and GPIO18(SDA)
00:00:00.059 TFS: File 'mcp23x.dat' not found
00:00:00.102 BRY: Berry initialized, RAM used 5719 bytes
00:00:01.017 DSP: Configured display 'ST7789'
00:00:01.024 Project tasmota - Boiler Version 13.3.0.2(lvgl-haspmota)-2_0_14(2024-01-02T13:49:51)
00:00:01.027 LVG: LVGL initialized
00:00:01.249 TAP: Loaded Tasmota App 'haspmota_widgets.tapp'
00:00:02.530 BRY: failed to run compiled code 'attribute_error' - 'nil' value has no writable attribute 'val'
00:00:04.787 WIF: Connecting to AP1 MyWiFiName Channel 1 BSSId D8:0D:17:C5:CF:17 in mode 11n as boilerdisplay-6140...
00:00:10.682 QPC: Reset
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x42034a4f  PS      : 0x00060930  A0      : 0x820cc738  A1      : 0x3fcb15b0  
A2      : 0x00000000  A3      : 0x3fcb598c  A4      : 0x0000020f  A5      : 0x3fcb5b9c  
A6      : 0x53504620  A7      : 0x007d657b  A8      : 0x820d7f1d  A9      : 0x3fcb1590  
A10     : 0x0000020f  A11     : 0x0000020f  A12     : 0x00000000  A13     : 0x000001f0  
A14     : 0x3fcb598c  A15     : 0x3fcb582b  SAR     : 0x00000017  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x000000fc  LBEG    : 0x40056f5c  LEND    : 0x40056f72  LCOUNT  : 0x00000000  

Backtrace: 0x42034a4c:0x3fcb15b0 0x420cc735:0x3fcb15d0 0x420cc752:0x3fcb15f0 0x420ce206:0x3fcb1610 0x420036df:0x3fcb1640 0x420cc4ed:0x3fcb1680 0x4209c245:0x3fcb16d0 0x42084872:0x3fcb1720 0x420cce26:0x3fcb1740 0x420de901:0x3fcb1760 0x4202a0c3:0x3fcb17d0

ELF file SHA256: 0000000000000000
```
The reboot occurs about 7 seconds after the QPC Reset command (?). In that time I can issue commands via the console, although I only tried ipaddress1 and it responds with the fixed address I set. It also displays the right things when I press either of the 2 onboard buttons:
```
00:00:12.057 RSL: RESULT = {"Button1":{"Action":"SINGLE"}}
00:00:12.060 GPIO0 button pressed -  SINGLE Button1#Action
00:00:19.979 RSL: RESULT = {"Button2":{"Action":"SINGLE"}}
```
The middle line there being from my Berry print() command.

Any clues on why this has happened and what I should do?

**Comment by s-hadinger** [comment]
There are so many things that could cause this. Did you compile yourself? Do you have the `.map` file?

**Comment by richardsg307** [comment, marked as answer]
I just rebooted my Router and it's all working again. Phew. I do sometimes get a connection problem but never before with the WiFi Arcs showing it has connected!
No I didn't compile myself. I used the Tasmota32-S3 (QIO Flas/OPI PSRAM) from the web installer. Then Autoconfigured with the Lilygo S3 Display. Don't know what a `.map` file is!
I'm about to install it all now. Will do a write up tomorrow. Thanks for all your help along the way.

**Reply by sfromis** [reply]
I've also got the T-Display S3 with the same binary and auto-configuration. Seen no trouble, but I have no inputs (buttons or sensors) configured, nor any Berry code running.

And since you get an error message, I'd generally suggest to fix this, even if I'm not aware of a connection to the crash.
```
00:00:01.249 TAP: Loaded Tasmota App 'haspmota_widgets.tapp'
00:00:02.530 BRY: failed to run compiled code 'attribute_error' - 'nil' value has no writable attribute 'val'
```
With the error, it may not be useful to have that `.tapp` file there.

One general thing to make sure is that you have a good and stable power supply.

The '.map' file is part of the result of doing a build, but since you're using a precompiled build you do not have that.

**Comment by richardsg307** [comment]
I will try downloading the `.tapp` file again in case it's corrupt. Or I could remove the page that uses it from my application.

Now I've moved it to the boiler location it's not connecting to WiFi at all. Pity, it was all looking good.
I read that WiFi on the T-Display-S3 board is pretty poor so I'm working on an ESP 2866 WiFi Extender. I used to have one for another device but lost track of that or overwritten it. 
I've also ordered a WiFi antenna but not sure if I need to remove a jumper or something. Can't find anything online about it but seen elsewhere that you should 'disable' the onboard antenna.

**Reply by sfromis** [reply]
The usual procedure with those optional external antenna is that there is a 3-pad L-shaped setup, where you can twist a resistor 90 degrees to connect the corner to the free pad. Looks a lot like what I see on the board:
![t-display-s3-antenna](https://github.com/arendst/Tasmota/assets/47082390/42ccb2dd-0d76-4297-84ef-0674bfc643c8)
It is most probably a 0-ohm resistor, basically like a wire piece suitable for SMD, meaning that it is no disaster if you lose the "grain of sand", and connect by other means.

**Reply by barbudor** [reply]
Generally a blob of solder is fine

**Reply by sfromis** [reply]
If it gets a mess soldering on those small pads, and all 3 gets covered by the blob of solder, you could also cut the trace up (in the image) to the PCB antenna and call it a day.

**Reply by richardsg307** [reply]
Thanks for the advice. Late last night in desperation I ordered from Amazon Prime an antenna and a cheap Tapo WiFi Extender. They came today and I thought I'd try the Extender first. It works great and now it's all installed and working.
Now I have the task of changing all my yaml MQTT and Template sensors since the data is now coming in as a Tasmota device. So still a little bit of work to do.

**Reply by sfromis** [reply]
What I generally do is that I configure my own names (including the Topic), meaning that it is easy to swap another device to use the existing topic, if needed. I'd actually recommend against relying on the default naming setup, which is mainly there to avoid problems if more than one device gets connected (with same names) without being suitably configured to avoid this.

**Reply by sfromis** [reply]
BTW. You could also config a Tasmota device to act as a range extender, without needing some other device type. I've found this to work great, as long as I keep realistic expectation for speed and range. Still, internet browsing (including images) from a tablet worked fine through the Tasmota [Range Extender](https://tasmota.github.io/docs/Range-Extender/) feature.

**Reply by richardsg307** [reply]
I get what you are saying about names. I might be able to 'Template' the new sensors into the old ones (or some at least).
I did wonder if Tasmota could do range extending but didn't follow it up. The Tapo one was only £15 and just plugs into a socket and looks smart.
I've just discovered Home Assistant is now showing my 'boiler' as 2 separate devices with different MACs. The Temperatures are in one and the Switches indicating Demand and Burn are in another together with a Control for turning the display on/off. Both show an ANALOG Range1 sensor, which has been in the configuration from the outset but I don't know why (the Web GUI also shows a Frame Rate for some reason). It's not a big deal, just puzzling!

**Reply by richardsg307** [reply]
Where's the best place to do a write up of my Boiler monitor and display project?

**Reply by barbudor** [reply]
À discussion in the 'Show and Tell' category is great for that
